.github/scripts/trymerge.py

```diff
@@ -725,6 +742,29 @@ def get_changed_files(self) -> List[str]:
             raise RuntimeError("Changed file count mismatch")
         return self.changed_files
 
+    def get_submodules(self) -> List[str]:
+        if self.submodules is None:
+            rc = gh_graphql(GH_GET_REPO_SUBMODULES,
+                            name=self.project,
+                            owner=self.org)
+            info = rc["data"]["repository"]["submodules"]
+            self.submodules = [s["path"] for s in info["nodes"]]
+        return self.submodules
+
+    def get_changed_submodules(self) -> List[str]:
+        submodules = self.get_submodules()
+        return [f for f in self.get_changed_files() if f in submodules]
+
+    def has_valid_submodule_updates(self) -> bool:
```

Comment:
nit: feels weird for this to return true when there are zero submodule updates.

Might be better to rename it to `has_invalid_submodule_updates`

Reply:
Sure